qa/rpc-tests/wallet_z_sendmany.py

```diff
@@ -164,9 +171,29 @@ def run_test(self):
 
         # Change went to a fresh address, so use `ANY_TADDR` which
         # should hold the rest of our transparent funds.
+        source = 'ANY_TADDR'
         recipients = []
         recipients.append({"address":n0ua0, "amount":10})
-        opid = self.nodes[2].z_sendmany('ANY_TADDR', recipients, 1, 0)
+
+        # If we attempt to spend with the default privacy policy, z_sendmany
+        # fails because it needs to spend transparent coins in a transaction
+        # involving a Unified Address.
+        revealed_senders_msg = 'This transaction requires selecting transparent coins, which is not enabled by default because it will publicly reveal transaction senders and amounts. THIS MAY AFFECT YOUR PRIVACY. Resubmit with the `privacyPolicy` parameter set to `AllowRevealedSenders` or weaker if you wish to allow this transaction to proceed anyway.'
+        opid = self.nodes[2].z_sendmany(source, recipients, 1, 0)
+        wait_and_assert_operationid_status(self.nodes[2], opid, 'failed', revealed_senders_msg)
+
+        # If we set any policy that does not include AllowRevealedSenders,
+        # z_sendmany also fails.
+        for policy in [
+            'FullPrivacy',
+            'AllowRevealedAmounts',
+            'AllowRevealedRecipients',
+        ]:
+            opid = self.nodes[2].z_sendmany(source, recipients, 1, 0, policy)
+            wait_and_assert_operationid_status(self.nodes[2], opid, 'failed', revealed_senders_msg)
+
+        # By setting the correct policy, we can create the transaction.
```

Comment:
Test that setting a weaker policy works.

Also, test that using an unrecognised policy name fails.

Reply:
I have that test further down in the file:
```
        # Send funds back from the legacy taddr to the UA. This requires
        # AllowRevealedSenders, but we can also use any weaker policy that
        # includes it.
```

Reply:
You check AllowFullyTransparent, but not NoPrivacy.

I know it's somewhat inconvenient to check multiple cases because each successful transaction affects the balances, but I think it would be worth it to be systematic.

Reply:
Guess it's another late night for me then...